### PR TITLE
Client.bearerToken setter

### PR DESCRIPTION
**Pull request by foxtree**
Trello card:

https://trello.com/c/BrTEC641/18-set-bearer-header

Note that the ^ original Trello card asks for a "setter"

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -75,6 +75,16 @@ class Client {
...
   }
 
+  /**
+   * Set the axios Authorization header to `Bearer ${token}`.
+   *
+   * @param {string} token The access token value.
+   */
+  // eslint-disable-next-line class-methods-use-this
+  set accessToken(token) {
+    axios.defaults.headers.common.Authorization = `Bearer ${token}`;
```
Comment (foxtree) [inline lib/client.js]: Not sure about this -- with ^this implementation, are the headers scoped *only* to our fhir client instance?

Either way: the same problem exists with the standalone `httpGet` function that is closed-over.  (which also resets the "Accept" header on every invocation)

File: test/client_test.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ describe('Client', () => {
     beforeEach(function () {
       nock(baseUrl)
         .matchHeader('accept', 'application/json+fhir')
+        .matchHeader('Authorization', '')
         .get('/metadata')
```
Comment (foxtree) [inline test/client_test.js]: This is appropriate, here, because requests for `/metadata` should work without authorization?

Comment (Jammjammjamm) [reply test/client_test.js]: `/metadata` is where you get the necessary OAuth urls, so it needs to be accessible without authorization.

Comment (bkaney) [reply test/client_test.js]: We need to remove this before merging.

**Comment by bkaney** [comment]
Also, does this set it on the class? We need a way to set it only for the instance of a client. Accept and Content-Type headers can be set on the class, but *not* the Authorization header.

File: test/client_test.js (hunk 2 above)
Comment (foxtree) [reply test/client_test.js]: This is testing that the 'Authorization' header is empty, when requesting `/metadata`

Comment (foxtree) [reply test/client_test.js]: This is testing that the 'Authorization' header is *empty*, when requesting `/metadata`

File: test/client_test.js [hunk 3]
```diff
@@ -43,6 +43,7 @@ describe('Client', () => {
       it('returns SMART OAuth URIs', async function () {
         nock(this.baseUrl)
           .matchHeader('accept', 'application/json+fhir')
+          .matchHeader('Authorization', '')
```
Comment (foxtree) [inline test/client_test.js]: To reiterate: this line tests that requests for `/metadata` have an *empty* Authorization header...

Comment (foxtree) [inline test/client_test.js]: To reiterate: this line tests that requests for `/metadata` can work with an *empty* Authorization header...

Comment (foxtree) [reply test/client_test.js]: (and: it is empty, by default)

File: lib/client.js [hunk 4]
```diff
@@ -134,7 +139,7 @@ class Client {
    * @example
    *
    * // Using promises
-   * fhirClient.search({ resourceType: 'Patient',
+   * fhirClient.search({ resourceType: 'Pataient',
```
Comment (rserur) [inline lib/client.js]: Looks like a typo with 'Patient' here

File: test/client-test.js [hunk 5]
```diff
@@ -43,6 +43,7 @@ describe('Client', () => {
       it('returns SMART OAuth URIs', async function () {
         nock(this.baseUrl)
           .matchHeader('accept', 'application/json+fhir')
+          .matchHeader('Authorization', '')
```
Comment (bkaney) [inline test/client-test.js]: I don't think we send a value-less header here.

Comment (foxtree) [reply test/client-test.js]: This assertion would still pass, in `origin/master`

Comment (bkaney) [reply test/client-test.js]: okay, then let's 👍  this PR and fix it later.

**Pull request merged**